src/components/code-editor/code-editor.tsx

```diff
@@ -0,0 +1,140 @@
+import {
+    Component,
+    h,
+    Prop,
+    Element,
+    Event,
+    EventEmitter,
+} from '@stencil/core';
+import { Language } from './code-editor.types';
+import CodeMirror from 'codemirror';
+import 'codemirror/mode/javascript/javascript';
+import 'codemirror/addon/selection/active-line';
+
+/**
+ * @exampleComponent limel-example-code-editor
+ */
+@Component({
+    tag: 'limel-code-editor',
+    shadow: true,
+    styleUrl: 'code-editor.scss',
+})
+export class CodeEditor {
+    /**
+     * The code to be rendered
+     */
+    @Prop()
+    public value: string = '';
+
+    /**
+     * The language of the code
+     */
+    @Prop()
+    public language: Language;
+
+    /**
+     * Disables editing of the editor content
+     */
+    @Prop()
+    public readonly: boolean = false;
+
+    /**
+     * Disables editing of the editor content
+     */
+    @Prop()
+    public lineNumbers: boolean = false;
+
+    /**
+     * Emitted when the code has changed. Will only be emitted when the code
+     * area has lost focus
+     */
+    @Event()
+    public change: EventEmitter<string>;
+
+    @Element()
+    private host: HTMLLimelCodeEditorElement;
+
+    private editor: CodeMirror.Editor;
+    private observer: IResizeObserver;
+
+    public connectedCallback() {
+        this.observer = new ResizeObserver(this.handleResize) as any;
+        this.observer.observe(this.host);
+    }
+
+    public disconnectedCallback() {
+        this.observer.unobserve(this.host);
+        this.editor.off('change', this.handleChange);
+        this.editor = null;
+    }
+
+    public componentDidRender() {
+        if (this.editor) {
+            return;
+        }
+
+        this.editor = this.createEditor();
+    }
+
+    private handleChange = () => {
+        this.change.emit(this.editor.getValue());
+    };
+
+    private handleResize = () => {
+        if (!this.editor) {
+            return;
+        }
+
+        this.editor.refresh();
+    };
+
+    private createEditor() {
+        const options = this.getOptions();
+
+        const editor = CodeMirror(
+            this.host.shadowRoot.querySelector('.editor'),
+            options
+        );
+
+        editor.on('change', this.handleChange);
+
+        return editor;
+    }
+
+    private getOptions(): CodeMirror.EditorConfiguration {
+        let mode: string | object = this.language;
+        const TAB_SIZE = 4;
+        let theme = 'solarized light';
+
+        if (matchMedia('(prefers-color-scheme: dark)').matches) {
+            theme = 'solarized dark';
+        }
```

Comment:
I did not know what theme to pick here, none of them seems to match the colors we have in Lime Elements at the moment. It should be quite easy to either just switch to another one or tweak the colors ourselves in the CSS @Kiarokh 

The ones available by default are these https://codemirror.net/demo/theme.html

Reply:
really nice! 

I prefer to edit the CSS if possible. Like what we did for Kompendium. This way we will have support for light and dark modes. Can we do it?

Reply:
I think you can merge this anyway. And we do another PR for the styles.

Reply:
Yeah I import the CSS files from Code Mirror in `code-editor.scss`, we can probably just override stuff in there or copy the contents of the imported files and change them manually if that is easier

Reply:
Cool. I prefer to copy the content, instead of overriding them. Do you want me to do it now or in a new PR?

Reply:
Either is fine with me. It seems no one has started reviewing this yet anyway :smile:

Reply:
OK I'll see if I can start before anyone reviews this